overlord/snapstate/autorefresh_test.go

```diff
@@ -1171,3 +1171,36 @@ func (s *autoRefreshTestSuite) TestTooSoonError(c *C) {
 	c.Check(snapstate.TooSoonError{}, Not(testutil.ErrorIs), errors.New(""))
 	c.Check(snapstate.TooSoonError{}.Error(), Equals, "cannot auto-refresh so soon")
 }
+
+func setStoreAccess(s *state.State, access string) {
+	s.Lock()
+	defer s.Unlock()
+
+	tr := config.NewTransaction(s)
+	tr.Set("core", "store.access", access)
+	tr.Commit()
+}
+
+func (s *autoRefreshTestSuite) TestSnapStoreOffline(c *C) {
+	s.addRefreshableSnap("foo")
+
+	setStoreAccess(s.state, "offline")
+
+	af := snapstate.NewAutoRefresh(s.state)
+	err := af.Ensure()
+	c.Check(err, IsNil)
+
+	s.state.Lock()
+	c.Check(s.state.Changes(), HasLen, 0)
+	s.state.Unlock()
+
+	setStoreAccess(s.state, "")
```

Comment:
I really meant nil not "", does that no work?

Reply:
Fixed!